src/software/ai/hl/stp/stp_tactic_assignment_test.cpp

```diff
@@ -472,9 +476,42 @@ TEST_F(STPTacticAssignmentTest,
     auto goalie_tactic_2                         = std::make_shared<GoalieTestTactic>();
     std::vector<std::shared_ptr<Tactic>> tactics = {goalie_tactic_1, goalie_tactic_2};
 
-    stp.assignRobotsToTactics(world, tactics);
+    stp.assignRobotsToTactics(world, &tactics);
 
     ASSERT_TRUE(goalie_tactic_1->getAssignedRobot().has_value());
     EXPECT_FALSE(goalie_tactic_2->getAssignedRobot().has_value());
     EXPECT_EQ(goalie_tactic_1->getAssignedRobot().value(), robot_0);
 }
+
+TEST_F(STPTacticAssignmentTest,
+       test_assigning_stop_tactics_to_unassigned_non_goalie_robots)
+{
+    // Test that StopTactic is assigned to remaining robots without tactics
+
+    Team friendly_team(Duration::fromSeconds(0));
+    Robot robot_0(0, Point(-1, 1), Vector(), Angle::zero(), AngularVelocity::zero(),
+                  Timestamp::fromSeconds(0));
+    Robot robot_1(1, Point(-3, 5), Vector(), Angle::zero(), AngularVelocity::zero(),
+                  Timestamp::fromSeconds(0));
+    Robot robot_2(2, Point(6, 7), Vector(), Angle::zero(), AngularVelocity::zero(),
+                  Timestamp::fromSeconds(0));
+    friendly_team.updateRobots({robot_0, robot_1, robot_2});
+    world.updateFriendlyTeamState(friendly_team);
+
+    auto move_tactic_1 = std::make_shared<MoveTestTactic>();
+
+    move_tactic_1->updateControlParams(Point(-1, 0));
+
+    std::vector<std::shared_ptr<Tactic>> tactics = {move_tactic_1};
+    std::vector<Robot> expected_robots_assigned  = {robot_0, robot_1, robot_2};
+    std::vector<std::string> expected_tactics    = {"Move Test Tactic", "Stop Tactic",
+                                                 "Stop Tactic"};
```

Comment:
Can we try catch dynamic_cast<MoveTestIntent&> instead of comparing strings?

Reply:
you can also do `dynamic_pointer_cast` which will return a `nullptr` if the cast failed, which you can check directly with
```
auto tactic = dynamic_pointer_cast<SomeTactic>(result);
ASSERT_TRUE(tactic);
...
```
and it's shorter / easier to write than try-catch

Reply:
Attempting to do `auto tactic1 = dynamic_cast<MoveTestTactic&>(*move_tactic_1);` gives an error "Call to implicitly-deleted copy-constructor of 'MoveTestTactic'". The other two tactics are for `StopTactic` which is causing `dynamic_cast` to err about an incomplete type.

Reply:
Done.